src/main/java/seedu/address/logic/commands/EditMeetingCommand.java

```diff
@@ -0,0 +1,220 @@
+            throw new CommandException(Messages.MESSAGE_INVALID_MEETING_DISPLAYED_INDEX);
+        }
+
+        Meeting meetingToEdit = clientList.get(clientIndex.getZeroBased())
+                .getMeetings().get(meetingIndex.getZeroBased());
+        Meeting editedMeeting = createEditedMeeting(meetingToEdit, editMeetingDescriptor);
+
+        if (meetingToEdit.isSameMeeting(editedMeeting) && model.hasMeeting(editedMeeting)) {
+            throw new CommandException(MESSAGE_DUPLICATE_MEETING);
+        }
+
+        DeleteMeetingCommand deleteMeetingCommand = new DeleteMeetingCommand(clientIndex, meetingIndex);
+        AddMeetingCommand addMeetingCommand = new AddMeetingCommand(editedMeeting.getDateTime(),
+                editedMeeting.getDescription(), clientIndex);
+        deleteMeetingCommand.execute(model);
+        addMeetingCommand.execute(model);
+        return new CommandResult(String.format(MESSAGE_EDIT_MEETING_SUCCESS, Messages.formatMeeting(editedMeeting)));
+    }
+
+    /**
+     * Creates and returns a {@code Meeting} with the details of {@code meetingToEdit}
+     * edited with {@code editMeetingDescriptor}.
+     */
+    private static Meeting createEditedMeeting(Meeting meetingToEdit, EditMeetingDescriptor editMeetingDescriptor) {
+        assert meetingToEdit != null;
+
+        String updatedName = editMeetingDescriptor.getDescription().orElse(meetingToEdit.getDescription());
+        LocalDateTime updatedDateTime = editMeetingDescriptor.getDateTime().orElse(meetingToEdit.getDateTime());
+        Person updatedClient = editMeetingDescriptor.getClient().orElse(meetingToEdit.getClient());
+        return new Meeting(updatedName, updatedDateTime, updatedClient);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditMeetingCommand)) {
+            return false;
+        }
+
+        EditMeetingCommand otherEditMeeting = (EditMeetingCommand) other;
+        return meetingIndex.equals(otherEditMeeting.meetingIndex)
+                && editMeetingDescriptor.equals(otherEditMeeting.editMeetingDescriptor)
+                && clientIndex.equals(otherEditMeeting.clientIndex);
+    }
+
+    public Index getClientIndex() {
+        return clientIndex;
+    }
+
+    public Index getMeetingIndex() {
+        return meetingIndex;
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("clientIndex", clientIndex)
+                .add("editMeetingDescriptor", editMeetingDescriptor)
+                .add("meetingIndex", meetingIndex)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the meeting with. Each non-empty field value will replace the
+     * corresponding field value of the meeting.
+     */
+    public static class EditMeetingDescriptor {
```

Comment:
interesting way to approach the problem, could you explain your thought process for creating a static class and using optionals?